store/sqlstore/post_store.go

```diff
@@ -2598,19 +2601,36 @@ func (s *SqlPostStore) permanentDeleteThreads(transaction *sqlxTxWrapper, postId
 	return nil
 }
 
-// Thread cleanup upon post deletion
-// if the post is a comment
-// reply count is reduced by 1 and,
-// the user is removed from participants if the comment deleted is the last reply from said user.
-func (s *SqlPostStore) cleanupThreadComments(transaction *sqlxTxWrapper, postId, rootId string, userId string) error {
+// deleteThread marks a thread as deleted at the given time.
+func (s *SqlPostStore) deleteThread(transaction *sqlxTxWrapper, postId string, deleteAtTime int64) error {
+	queryString, args, err := s.getQueryBuilder().
+		Update("Threads").
+		Set("DeleteAt", deleteAtTime).
+		Where(sq.Eq{"PostId": postId}).
+		ToSql()
+	if err != nil {
+		return errors.Wrapf(err, "failed to create SQL query to mark thread for root post %s as deleted", postId)
+	}
+
+	_, err = transaction.Exec(queryString, args...)
+	if err != nil {
+		return errors.Wrapf(err, "failed to mark thread for root post %s as deleted", postId)
+	}
```

Comment:
`deleteThread` is always called when `id.RootId == ""` when deleting a post. If the post isn't a thread, will that result in an error?

Reply:
Scanning the code, all the `UPDATE`s should complete without error even if the affected rows are empty. But for completeness, I reviewed (and refactored for clarity) the test coverage for deleting a post and verified that this case is covered (root post without a reply, root post with one reply, root post with multiple replies). No errors are emitted when there is no thread.